lib/rules/require-unicode-regexp.js

```diff
@@ -85,22 +139,34 @@ module.exports = {
                     const pattern = getStringIfConstant(patternNode, scope);
                     const flags = getStringIfConstant(flagsNode, scope);
 
-                    if (!flagsNode || (typeof flags === "string" && !flags.includes("u") && !flags.includes("v"))) {
+                    let missingFlag = !flagsNode;
+
+                    if (typeof flags === "string") {
+                        missingFlag = checkFlags(requireFlag, flags);
+                    }
+
+                    if (missingFlag) {
                         context.report({
-                            messageId: "requireUFlag",
+                            messageId: requireFlag === "v" ? "requireVFlag" : "requireUFlag",
                             node: refNode,
-                            suggest: typeof pattern === "string" && isValidWithUnicodeFlag(context.languageOptions.ecmaVersion, pattern)
+                            suggest: typeof pattern === "string" && isValidWithUnicodeFlag(context.languageOptions.ecmaVersion, pattern, requireFlag)
                                 ? [
                                     {
                                         fix(fixer) {
+                                            const replaceFlag = requireFlag ?? "u";
+
                                             if (flagsNode) {
                                                 if ((flagsNode.type === "Literal" && typeof flagsNode.value === "string") || flagsNode.type === "TemplateLiteral") {
                                                     const flagsNodeText = sourceCode.getText(flagsNode);
+                                                    const flag = requireFlag === "u" ? "v" : "u";
 
                                                     return fixer.replaceText(flagsNode, [
-                                                        flagsNodeText.slice(0, flagsNodeText.length - 1),
+                                                        flagsNodeText.slice(0, flagsNodeText.length - 1).replace(
+                                                            flag,
+                                                            ""
+                                                        ),
```

Comment:
In these cases, the fix wouldn't be correct:

```js
/*eslint require-unicode-regexp: ["error", { "requireFlag": "v" }] */

const foo = new RegExp("foo", "\u0067"); // fixed to "\0067v"

const regularFlags = "sm";

const bar = new RegExp("bar", `${regularFlags}g`); // fixed to `${reglarFlags}gv`

```

I think, in case when the other flag is already present, it's fine to provide the suggestion only if the flags node is a string literal without escape sequences (i.e. without a `\` character) or a template literal with no expressions and no escape sequences.

Reply:
Provided a fix, thanks.